libs/api/books/src/lib/books.service.ts

```diff
@@ -9,26 +9,28 @@ export class BooksService {
 
   search(term: string): Observable<Book[]> {
     if (!term) {
-      throw new Error('Missing serach term');
+      throw new Error('Missing search term');
     }
 
     return this.http
       .get(`https://www.googleapis.com/books/v1/volumes?q=${term}`)
       .pipe(
-        map(resp => {
-          return resp.data.items.map(item => {
-            return {
-              id: item.id,
-              title: item.volumeInfo?.title,
-              authors: item.volumeInfo?.authors || [],
-              description: item.searchInfo?.textSnippet,
-              publisher: item.volumeInfo?.publisher,
-              publishedDate: item.volumeInfo?.publishedDate
-                ? new Date(item.volumeInfo?.publishedDate).toISOString()
-                : undefined,
-              coverUrl: item.volumeInfo?.imageLinks?.thumbnail
-            };
-          });
+        map((resp) => {
+          if (resp.data.items) {
+            return resp.data.items.map((item) => {
+              return {
+                id: item.id,
+                title: item.volumeInfo?.title,
+                authors: item.volumeInfo?.authors || [],
+                description: item.searchInfo?.textSnippet,
+                publisher: item.volumeInfo?.publisher,
+                publishedDate: item.volumeInfo?.publishedDate
+                  ? new Date(item.volumeInfo?.publishedDate).toISOString()
+                  : undefined,
+                coverUrl: item.volumeInfo?.imageLinks?.thumbnail,
+              };
+            });
+          }
```

Comment:
Fetching the book from API and checking the condition if API response is not null then it will bind with the model using API result data.

Reply:
This one still throw error, if u get an empty response